src/core/ext/xds/xds_client.cc

```diff
@@ -1036,10 +1047,15 @@ void XdsClient::ChannelState::AdsCallState::OnRecvMessage(
     absl::string_view payload) {
   {
     MutexLock lock(&xds_client()->mu_);
-    if (!IsCurrentCallOnChannel()) return;
+    if (!IsCurrentCallOnChannel()) {
+      call_->StartRecvMessage();
```

Comment:
This is extremely counter-intuitive -- why would we need to start a new read on a stream that has been shut down?  I think the fact that you needed to add this is a symptom of the ref-counting problem I pointed out elsewhere.  If you make the changes I suggested to require the caller to explicitly start each read and to take a new ref for each read, then I don't think this call will be necessary.